src/query/interpreter.cpp

```diff
@@ -416,19 +421,30 @@ Callback HandleReplicationQuery(ReplicationQuery *repl_query, const Parameters &
   ExpressionEvaluator evaluator(&frame, symbol_table, evaluation_context, db_accessor, storage::View::OLD);
 
   Callback callback;
+  std::optional<Notification> replica_notification;
   switch (repl_query->action_) {
     case ReplicationQuery::Action::SET_REPLICATION_ROLE: {
       auto port = EvaluateOptionalExpression(repl_query->port_, &evaluator);
       std::optional<int64_t> maybe_port;
       if (port.IsInt()) {
         maybe_port = port.ValueInt();
       }
+      if (maybe_port == 7687 && repl_query->role_ == ReplicationQuery::ReplicationRole::REPLICA) {
+        const auto replica_port_notification =
+            Notification(SeverityLevel::WARNING, NotificationCode::REPLICA_PORT_WARNING,
+                         "Be careful the replication port must be different from the memgraph port!");
+        notifications->emplace_back(replica_port_notification.ConvertToMap());
+      }
       callback.fn = [handler = ReplQueryHandler{interpreter_context->db}, role = repl_query->role_,
                      maybe_port]() mutable {
         handler.SetReplicationRole(role, maybe_port);
         return std::vector<std::vector<TypedValue>>();
       };
-      return callback;
+      replica_notification =
+          Notification(SeverityLevel::INFO, NotificationCode::SET_REPLICA,
+                       fmt::format("Replica role set to {}.",
+                                   repl_query->role_ == ReplicationQuery::ReplicationRole::MAIN ? "MAIN" : "REPLICA"));
```

Comment:
why don't you emplace immediately like above and return callback?

Reply:
Because `replica_notification` is of type `optional<Notification>`, which before being inserted int notifications vector is transformed into the map by calling `ConvertToMap`, and it is convenient to do that right before returning callback, then in every place, call `ConvertToMap` (you can see line below the example above that `replica_port_notification` is being emplaced into notifications vector)

Reply:
I don't like it just because you put a constraint of having exact one Notification, while every extra notification is handled differently.
You can always do something like `notification.emplace_back(Notification(...).ConvertToMap())`.

Reply:
I was thinking of doing that but didn't since there is a lot of repetitive 'notifications.emplace_back(Notification(...).ConvertToMap())', and when using optional we just check that at the end. Maybe it makes sense to actually have a vector of Notifications and then put it into the summary at the end (converted into a map of course)?

Reply:
Yes, I think it is better to have a vector of Notifications and convert them only when necessary:
1. the query engine doesn't need to know how the notifications are communicated to the users
2. Spare a lot of `ConvertToMap()` calls
3. In the future we can easily reuse those notifications if we want to (e.g.: log them and the transaction is committed for some debugging reason etc.)

I wanted to mention this in my review, but somehow I managed to not write the comment. (I checked my comments, and nothing else seems to be missing.)